### PR TITLE
Feat / Geo block message home

### PR DESCRIPTION
**Pull request by royschut**
# Geo block message home
- Adds a geo block message error screen, whenever all first playlists are succesful yet empty - [OTT-360]
- Refactors `<ShelfList />` to use `useQueries()` instead of `<PlaylistContainer />`, so that we can verify if the queries have succeeded and contain content
- This made PlaylistContainer only used for the Favorites screen, so I removed it and simplified the Favorites component as well. The loading and error state weren't even valid, because after login they always get fetched and awaited.

## Question
- It's correct to merge this to the a11y branch?

[OTT-360]: https://videodock.atlassian.net/browse/OTT-360?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/ui-react/src/components/Shelf/Shelf.tsx [hunk 1]
```diff
@@ -137,8 +137,8 @@ const Shelf = ({
   if (error || !playlist?.playlist) return <h2 className={styles.error}>Could not load items</h2>;
 
   return (
-    <div className={classNames(styles.shelf, { [styles.featured]: featured })}>
-      {!featured ? <h2 className={classNames(styles.title, { [styles.loading]: loading })}>{title || playlist.title}</h2> : null}
+    <div className={classNames(styles.shelf)}>
+      {!featured ? <h2 className={classNames(styles.title)}>{title || playlist.title}</h2> : null}
```
Comment (royschut) [inline packages/ui-react/src/components/Shelf/Shelf.tsx:141]: They don't exist, obsolete

File: packages/ui-react/src/containers/ShelfList/ShelfList.module.scss [hunk 2]
```diff
@@ -7,14 +7,6 @@
...
 }
 
-.list {
-  overflow: visible !important;
-
-  > div {
-    overflow: visible !important;
-  }
-}
-
```
Comment (royschut) [inline packages/ui-react/src/containers/ShelfList/ShelfList.module.scss:17]: Wasn't used.

File: packages/ui-react/src/containers/ShelfList/ShelfList.tsx [hunk 3]
```diff
@@ -1,88 +1,136 @@
...
-  const [rowCount, setRowCount] = useState(INITIAL_ROW_COUNT);
+  const [rowsLoaded, setRowsLoaded] = useState(INITIAL_ROWS_LOADED);
+  const { t } = useTranslation('error');
 
   const watchHistoryDictionary = useWatchHistoryStore((state) => state.getDictionaryWithSeries());
 
   // User
   const { user, subscription } = useAccountStore(({ user, subscription }) => ({ user, subscription }), shallow);
 
+  // Todo: move to more common package?
```
Comment (royschut) [inline packages/ui-react/src/containers/ShelfList/ShelfList.tsx:39]: If someone has a good and relatively easy suggestion then I'm happy to do this now.

File: packages/ui-react/src/components/Shelf/Shelf.tsx (hunk 1 above)
Comment (ChristiaanScheermeijer) [reply packages/ui-react/src/components/Shelf/Shelf.tsx:141]: The classNames you mean?

File: packages/ui-react/src/components/Favorites/Favorites.tsx [hunk 4]
```diff
@@ -53,7 +42,7 @@ const Favorites = ({ playlist, error, isLoading, accessModel, hasSubscription, o
           playlist={playlist}
           onCardHover={onCardHover}
           cols={cols}
-          isLoading={isLoading}
+          isLoading={false}
```
Comment (ChristiaanScheermeijer) [inline packages/ui-react/src/components/Favorites/Favorites.tsx:45]: Aren't we losing the loading state (placeholder data)?

File: packages/ui-react/src/containers/ShelfList/ShelfList.tsx [hunk 5]
```diff
@@ -1,88 +1,136 @@
...
     // reset row count when the page changes
-    return () => setRowCount(INITIAL_ROW_COUNT);
+    return () => setRowsLoaded(INITIAL_ROWS_LOADED);
   }, [rows]);
 
+  // If all playlists are empty, due to geo restrictions, we show a geo block error page
+  const allPlaylistsEmpty = playlists.every(({ data, isSuccess }) => isSuccess && !data?.playlist?.length);
+
+  if (allPlaylistsEmpty) {
+    return <ErrorPage title={t('geo_blocked_heading')} message={t('geo_blocked_description')} />;
```
Comment (ChristiaanScheermeijer) [inline packages/ui-react/src/containers/ShelfList/ShelfList.tsx]: Minor 🐜 ; we assume that geo-blocking is the reason for the empty shelves. But it could be a misconfiguration as well. The translation is correct, but the key indicates that it is a geo-block error. 

```suggestion
    return <ErrorPage title={t('empty_shelves_heading')} message={t('empty_shelves_description')} />;
```

**Review by ChristiaanScheermeijer — APPROVED** [review]
🙌

File: packages/ui-react/src/containers/ShelfList/ShelfList.tsx (hunk 3 above)
Comment (ChristiaanScheermeijer) [reply packages/ui-react/src/containers/ShelfList/ShelfList.tsx:39]: We could encapsulate this in a `usePlaylists` or `useShelvesData` hook?

File: packages/ui-react/src/components/Favorites/Favorites.tsx (hunk 4 above)
Comment (royschut) [reply packages/ui-react/src/components/Favorites/Favorites.tsx:45]: For favorites? They are loaded and awaited during login (see PR decsr).

File: packages/ui-react/src/components/Shelf/Shelf.tsx (hunk 1 above)
Comment (royschut) [reply packages/ui-react/src/components/Shelf/Shelf.tsx:141]: Sorry, yes

File: packages/ui-react/src/containers/ShelfList/ShelfList.tsx [hunk 6]
```diff
@@ -1,88 +1,136 @@
...
-                      posterAspect={posterAspect}
-                      visibleTilesDelta={visibleTilesDelta}
-                    />
-                  </div>
-                </div>
-              );
-            }}
-          </PlaylistContainer>
-        ))}
+        {rows.slice(0, rowsLoaded - 1).map(({ type, featured, title }, index) => {
```
Comment (MelissaDTH) [inline packages/ui-react/src/containers/ShelfList/ShelfList.tsx]: We've discussed this but as a reminder: we want to show the amount of loaded rows. When we add `- 1` here, we will not show the last rendered row and therefore miss 1 row. We can remove the `-1` here!

File: packages/ui-react/src/containers/ShelfList/ShelfList.tsx (hunk 3 above)
Comment (royschut) [reply packages/ui-react/src/containers/ShelfList/ShelfList.tsx:39]: Yes, that would make it react (including react-native), perhaps that's enough for now? Easy implementation indeed.

File: packages/ui-react/src/containers/ShelfList/ShelfList.tsx (hunk 5 above)
Comment (royschut) [reply packages/ui-react/src/containers/ShelfList/ShelfList.tsx]: Good point. We do say "... in your region" in the translations. Are we sure this is sufficient then?

**Comment by MelissaDTH** [comment]
I'm not sure about merging this to the a11y branch, because I thought we want to present that branch as an accessibility package containing solely a11y changes. What were your thoughts on this?

**Comment by royschut** [comment]
> I'm not sure about merging this to the a11y branch, because I thought we want to present that branch as an accessibility package containing solely a11y changes. What were your thoughts on this?

Last sprint we had a `restructure` branch and an `a11y` branch which also includes other v1 features (only geo block at that time), that's why I did the same here. If it doesn't fit with a11y, then perhaps we need a 3rd working branch. @Videodock/videodock-front-end ?

**Comment by MelissaDTH** [comment]
> Last sprint we had a `restructure` branch and an `a11y` branch which also includes other v1 features (only geo block at that time), that's why I did the same here. If it doesn't fit with a11y, then perhaps we need a 3rd working branch. @Videodock/videodock-front-end ?

I understand where you're coming from. While introducing a third branch to maintain separation might be a good option, but if this feature would be the sole addition and we've already incorporated a v1 feature in the previous `a11y` branch, it might be excessive. If that's the case, I'm fine with merging it into the `a11y` branch!

**Review by MelissaDTH — APPROVED** [review]
LGTM!

**Comment by langemike** [comment]
Really nice approach! :D Thanks for picking this up.

Let's merge it with the `a11y` branch, because there are some a11y HTML optimisations planned regarding the `ShelfList`. Which makes it easier to apply it.

Some questions:
1. Have you thought about how we can easily test this on testday? Did you prepared some geo blocked content for a JW property?
2. What is the outcome if you have not defined any playlists for the homepage? Or you consider this an edge case?
3. I think it's also valuable to implement this for the Playlist page (PlayListGrid.tsx). What is your opinion about this?

Thanks!

**Comment by royschut** [comment]
> * Have you thought about how we can easily test this on testday? Did you prepared some geo blocked content for a JW property?
> * What is the outcome if you have not defined any playlists for the homepage? Or you consider this an edge case?
> * I think it's also valuable to implement this for the Playlist page (PlayListGrid.tsx). What is your opinion about this?

Great questions Mike!

1. The SVOD-Inplayer platform is fully geo-blocked. I've just communicated this to PM.
2. This will result into the same message. This looks better than the otherwise empty page, although "..in your region" is not correct in a way. However, I indeed think this is an edge case that doesn't need a solution
3. Yes, let's do it (see my last commit). You could argue that then also the menu items should be hidden, but I'm a bit afraid this leads to question from content editors. Keep it like this?

Screenshot 2024-01-25 at 08 02 05

**Review by langemike — APPROVED** [review]
> Keep it like this?

YES!

**Review by ChristiaanScheermeijer — APPROVED** [review]
Very nice 🙌 !
